src/win/i2c.cc

```diff
@@ -0,0 +1,318 @@
+  for (unsigned int i = 0; i < jsArray->Length(); i++) {
+    val = jsArray->Get(i);
+    bytes.push_back(static_cast<BYTE>(val->Uint32Value()));
+    Nan::Set(arr, i, val);
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  try {
+    obj->_i2cDevice->Write(ArrayReference<BYTE>(bytes.data(), static_cast<unsigned int>(bytes.size())));
+  } catch (Exception^ e) {
+    Nan::ThrowError(Nan::ErrnoException(e->HResult, "WinI2c::Write", ""));
+  }
+}
+
+NAN_METHOD(WinI2c::WritePartial) {
+  if (info.Length() < 2 ||
+      !info[0]->IsInt32() ||
+      !info[1]->IsObject()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::WritePartial",
+      "incorrect arguments passed to WinI2c::WritePartial"
+      "(int length, Buffer buffer)"));
+  }
+  
+  Nan::HandleScope scope;
+  uint32 length = info[0]->Uint32Value();
+  v8::Local<v8::Object> bufferHandle = info[1].As<v8::Object>();
+  int bytesWritten = -1;
+  byte* bufferData = (byte*)node::Buffer::Data(bufferHandle);
+  size_t bufferLength = node::Buffer::Length(bufferHandle);
+
+  if (length > bufferLength) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::WritePartial",
+      "buffer passed to WinI2c::WritePartial contains less than 'length' bytes"));
+  }
+
+  auto writeBuf = ref new Platform::Array<BYTE>(length);
+
+  for (int i = 0; i < writeBuf->Length; i++) {
+    writeBuf->set(i, bufferData[i]);
+  }
+
+  WinI2c* obj = Nan::ObjectWrap::Unwrap<WinI2c>(info.This());
+  try {
+    I2cTransferResult result = obj->_i2cDevice->WritePartial(writeBuf);
+    switch (result.Status) {
+      case I2cTransferStatus::FullTransfer:
+        bytesWritten = result.BytesTransferred;
+        break;
+      case I2cTransferStatus::PartialTransfer:
+        bytesWritten = result.BytesTransferred;
+        wprintf(L"%d bytes partially transferred\n", bytesWritten);
+        break;
+      case I2cTransferStatus::SlaveAddressNotAcknowledged:
+        wprintf(L"Slave address was not acknowledged\n");
+        break;
+      default:
+        wprintf(L"Invalid transfer status value\n");
+    }
+  }
+  catch (Exception^ e) {
+    Nan::ThrowError(Nan::ErrnoException(e->HResult, "WinI2c::WritePartial", ""));
+  }
+
+  if (bytesWritten == -1) {
+    return Nan::ThrowError(Nan::ErrnoException(errno, "WinI2c::WritePartial", ""));
+  }
+
+  info.GetReturnValue().Set(Nan::New<v8::Integer>(bytesWritten));
+}
+
+NAN_METHOD(WinI2c::WriteRead) {
+  Nan::ThrowError(Nan::ErrnoException(EPERM, "WinI2c::WriteRead", "Not Implemented"));
+}
+
+NAN_METHOD(WinI2c::WriteReadPartial) {
+  if (info.Length() < 3 ||
+      !info[0]->IsInt32() ||
+      !info[1]->IsInt32() ||
+      !info[2]->IsObject()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "WinI2c::WriteReadPartial",
+      "incorrect arguments passed to WinI2c::WriteReadPartial"
+      "(int cmd, int length, Buffer buffer)"));
+  }
+
+  Nan::HandleScope scope;
```

Comment:
An implicit HandleScope is created on JavaScript accessible methods. Why is another one created here?